src/coreclr/jit/valuenum.cpp

```diff
@@ -10379,38 +10384,37 @@ static bool GetObjectHandleAndOffset(ValueNumStore*         vnStore,
                                      ssize_t*               byteOffset,
                                      CORINFO_OBJECT_HANDLE* pObj)
 {
-
     if (!tree->gtVNPair.BothEqual())
     {
         return false;
     }
 
     ValueNum  treeVN = tree->gtVNPair.GetLiberal();
     VNFuncApp funcApp;
-    if (vnStore->GetVNFunc(treeVN, &funcApp) && (funcApp.m_func == (VNFunc)GT_ADD))
+    size_t    offset = 0;
```

Comment:
It is better to use `target_ssize_t` for `offset` (and `ConstantValue<target_ssize_t>` accordingly) to get the unambiguously right overflow behavior.

Reply:
Good point, I always forget about that